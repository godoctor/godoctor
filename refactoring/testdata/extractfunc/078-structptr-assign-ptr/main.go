package main

import "fmt"

func main() {
	s := &struct {
		member int
	}{}
	s = &struct{ member int }{4} // <<<<<extract,9,1,10,1,extracted,pass
	fmt.Println(s.member)
}
