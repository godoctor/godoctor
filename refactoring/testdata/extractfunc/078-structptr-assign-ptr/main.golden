package main

import "fmt"

func main() {
	s := &struct {
		member int
	}{}
	s = extracted() // <<<<<extract,9,1,10,1,extracted,pass
	fmt.Println(s.member)
}

func extracted() *struct{ member int } {
	var s *struct{ member int }
	s = &struct{ member int }{4}
	return s
}
